hazelcast/src/main/java/com/hazelcast/cluster/Address.java

```diff
@@ -150,6 +153,7 @@ public void readData(ObjectDataInput in) throws IOException {
         port = in.readInt();
         type = in.readByte();
         host = in.readUTF();
+        socketAddr = new InetSocketAddress(InetAddress.getByName(host), port);
```

Comment:
Might affect the performance if we would do frequent DNS requests here. Not sure how frequent they are in fact.

Reply:
:+1: I will change it.

Let's use an `InetAddress` instance instead of the `InetSocketAddress` and serialize it too.